zuul-core/src/main/java/com/netflix/zuul/netty/connectionpool/DefaultClientChannelManager.java

```diff
@@ -225,6 +238,8 @@ public boolean release(final PooledConnection conn) {
         discoveryResult.incrementNumRequests();
 
         if (shuttingDown) {
+            LOG.debug("[{}] closing connection released during shutdown", conn.getChannel().id());
+            conn.close();
```

Comment:
An explicit close is not necessary here. The graceful shutdown should handle closing the connections.

Reply:
I agree that would be better. I was working with the existing logic of not returning the connection to the pool during shutdown, but I'm not sure we need to keep doing that

Reply:
Yeah, `close()` is best delegated to the conn. pool itself.